### PR TITLE
Turn on closure compiler collapse properties.

### PR DESCRIPTION
**Pull request by cramforce**
Boom.

### CODE REVIEW

File: test/size.txt [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
-132.56 kB   |    24.46 kB   |    8.75 kB   |   v0/amp-twitter-0.1.js
-280.28 kB   |    63.14 kB   |   18.56 kB   |   v0/amp-user-notification-0.1.js
- 71.14 kB   |    10.34 kB   |    3.99 kB   |   v0/amp-vimeo-0.1.js
- 70.67 kB   |    10.21 kB   |    3.95 kB   |   v0/amp-vine-0.1.js
- 81.63 kB   |    13.23 kB   |    4.91 kB   |   v0/amp-youtube-0.1.js
-135.66 kB   |    36.85 kB   |   12.34 kB   |   current-min/f.js / current/integration.js
\ No newline at end of file
+      max   |         min   |               gzip   |   file
+      ---   |         ---   |                ---   |   ---
+  9.85 kB   |     3.82 kB   |          alp-v0.js
```
Comment (jridgewell) [inline test/size.txt]: Can you re-run sizes?

Comment (cramforce) [inline test/size.txt]: On it. Need to run `gulp clean` Takes a while.

Comment (cramforce) [inline test/size.txt]: The metrics are correct, though.

**Comment by erwinmombay** [comment]
@cramforce still seeing the disconnect i was seeing in https://github.com/ampproject/amphtml/pull/2951 with collapsProperties on v0.js (which is why i excluded it). didnt want to merge it and break testing for everybody

**Comment by cramforce** [comment]
Alright, excited to see what travis says. Worked around the `PJs` issue.

**Comment by cramforce** [comment]
Green build :)

**Comment by erwinmombay** [comment]
@cramforce LGTM!

**Comment by cramforce** [comment]
This will need some baking :)

**Comment by erwinmombay** [comment]
@cramforce good timing since we're only doing canary next week

**Comment by cramforce** [comment]
+1

File: test/size.txt [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
- 71.21 kB   |    10.43 kB   |    4.03 kB   |   v0/amp-soundcloud-0.1.js
- 79.18 kB   |    12.93 kB   |    4.67 kB   |   v0/amp-springboard-player-0.1.js
-132.56 kB   |    24.46 kB   |    8.75 kB   |   v0/amp-twitter-0.1.js
-280.28 kB   |    63.14 kB   |   18.56 kB   |   v0/amp-user-notification-0.1.js
- 71.14 kB   |    10.34 kB   |    3.99 kB   |   v0/amp-vimeo-0.1.js
- 70.67 kB   |    10.21 kB   |    3.95 kB   |   v0/amp-vine-0.1.js
- 81.63 kB   |    13.23 kB   |    4.91 kB   |   v0/amp-youtube-0.1.js
-135.66 kB   |    36.85 kB   |   12.34 kB   |   current-min/f.js / current/integration.js
\ No newline at end of file
+614.81 kB   |   157.73 kB   |   43.68 kB   |   v0.js / amp.js
```
Comment (kristoferbaxter) [inline test/size.txt]: Looking good here if the results are stable. ~9% reduction.

**Pull request merged**